java-checks/src/main/java/org/sonar/java/checks/StringLiteralDuplicatedCheck.java

```diff
@@ -65,6 +65,16 @@ public void scanFile(JavaFileScannerContext context) {
     scan(context.getTree());
     for (String entry : occurrences.keySet()) {
       Collection<LiteralTree> literalTrees = occurrences.get(entry);
+      boolean allInConstants = context.getSemanticModel() != null && literalTrees.stream().allMatch(t -> {
```

Comment:
I think it is simpler to ignore "constants" during the visit, i.e. cut the visit in the ``visitVariable`` method, so constants are not counted in ``occurences``